src/Reexport.jl

```diff
@@ -1,11 +1,21 @@
 module Reexport
 
-macro reexport(ex)
-    isa(ex, Expr) && (ex.head == :module ||
-                      ex.head == :using ||
-                      (ex.head == :toplevel &&
-                       all(e->isa(e, Expr) && e.head == :using, ex.args))) ||
-        error("@reexport: syntax error")
+macro reexport(ex::Expr)
+    esc(reexport(__module__, ex))
+end
+
+function reexport(m::Module, ex::Expr)
+    # unpack any macros
+    ex = macroexpand(m, ex)
+    # recursively unpack any blocks
+    if ex.head == :block
+        ex = Base.remove_linenums!(ex)
+        return Expr(:block, map(e -> reexport(m, e), ex.args)...)
+    end
+
+    Meta.isexpr(ex, [:module, :using, :import]) ||
+    Meta.isexpr(ex, :toplevel) && all(e->isa(e, Expr) && e.head == :using, ex.args) ||
+    error("@reexport: syntax error")
```

Comment:
```suggestion
        Meta.isexpr(ex, :toplevel) && all(e->isa(e, Expr) && e.head == :using, ex.args) ||
        error("@reexport: syntax error")
```
Minor style nit. I find it a bit more difficult to follow when subsequent conditions/side effects aren't indented after the initial condition.

Reply:
I just formatted the whole file with JuliaFormatter.jl. That fixed this formatting plus a few other minor things. Let me know if that formatting is too much, then I'll revert it and only do the changes you suggested here.

Reply:
Yeah, it's uhhhh a bit much 😅

JuliaFormatter with `YASStyle()` would be closer to the existing style used here (though not exactly the same).

Reply:
I reverted it and manually indented the things as requested (I hope). Let me know if it is what you want now.

Reply:
Yep, looks great, thanks!